test/test_common/proto_matchers.h

```diff
@@ -0,0 +1,72 @@
+#include <string>
+
+#include "external/envoy/source/common/protobuf/protobuf.h"
+#include "external/envoy/source/common/protobuf/well_known.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+// A custom IgnoreCriteria that can be added to a MessageDifferencer to ignore
+// unknown fields by their field number, regardless of where they appear in a
+// message.
+class IgnoreUnknownFieldsGloballyByNumber
```

Comment:
Just fyi I remember we may have standard proto comparing util function that can ignore certain fields, why do we want to implement our own?

Reply:
Thank you @qqustc, can you point me to the function you are referring to? I have implemented our own, because I didn't find any.

Reply:
ah, the one I remembered are internal so I guess we have to implement our own

Reply:
That was my conclusion as well. Thank you for bringing it up though.